File: algorithms/QuickSort.java
package algoritms;

//chosse pivot element
//a <= pivot <= c
//
public class QuickSort {

    public static void main(String[] args) {
        int[] numbers = new int[] { 32, 21, 90, 11, 60, 26, 9, 18, 7 };
        int lb = 0;
        int ub = numbers.length - 1;

        quickSort(numbers, lb, ub);

        for (int i = 0; i < numbers.length; i++) {
            System.out.print(numbers[i] + " ");
        }
    }

    public static void quickSort(int[] numbers, int lb, int ub) {

        if (lb < ub) {
            int index = partition(numbers, lb, ub);
            quickSort(numbers, lb, index - 1);
            quickSort(numbers, index + 1, ub);

        }

    }

    public static int partition(int[] numbers, int lb, int ub) {
        int pivot = numbers[lb];
        int start = lb;
        int end = ub;

        while (start < end) {
            while (pivot >= numbers[start] && start < end) {
                start++;
            }
            while (pivot < numbers[end]) {
                end--;
            }
            if (start < end) {
                int temp = numbers[start];
                numbers[start] = numbers[end];
                numbers[end] = temp;
            }
        }      
        numbers[lb] = numbers[end];
        numbers[end] = pivot;
        return end;
    }

}

File: problemsolving/Quick Sort.cpp
//Program for quick sort using pivot element
#include<iostream>
using namespace std;
int quickSort(int*,int,int);
void swap(int *a,int *b)
{
	int temp = *a;
	*a = *b;
	*b = temp;
}
void part(int a[],int lb,int ub)
{
	int l;
	if(lb<ub)
	{
		l = quickSort(a,lb,ub);
		part(a,lb,l-1);
		part(a,l+1,ub);
	}
}
int quickSort(int a[],int lb,int ub)
{
	int pivot,i,j;
	pivot = a[ub];
	i = lb-1;
	for(j=lb;j<ub;j++)
	{
		if(a[j]<=pivot)
		{
			i += 1;
			swap(&a[i],&a[j]);
		}
	}
	swap(&a[i+1],&a[ub]);
	return(i+1);
}
int main()
{
	int a[20],size,i;
	cout<<"Enter size of array :";
	cin>>size;
	cout<<"Enter array "<<endl;
	for(i=0;i<size;i++)
	{
		cin>>a[i];
	}
	part(a,0,size-1);
	cout<<"Sorted array is"<<endl;
	for(i=0;i<size;i++)
		cout<<a[i]<<"\t";
	return(0);
}
